lib/config.js

```diff
@@ -44,7 +44,8 @@ var DEFAULTS = {
   sourceLinesSpanLibraryFrames: 0,
   flushInterval: 10,
   transactionMaxSpans: Infinity,
-  transactionSampleRate: 1.0
+  transactionSampleRate: 1.0,
+  serverTimeout: Infinity
```

Comment:
After investigating an issue today with a socket that wasn't timed out properly by the APM Server, which resulted in the agent just opening more and more connections to the APM Server, I think it would make sense to choose a sensible default here, e.g. somewhere between 30 and 60 seconds. What do you think?

Reply:
Sounds good to me. I just went with Infinity because that was the *original* behaviour. Having an actually *reasonable* timeout is definitely a good idea though.

Reply:
Good point regarding original behavior... while on one hand this could be considered a breaking change, as long as we don't choose a default that's too low, then you could also consider it a bug fix 😅